### PR TITLE
Feature setup socket io #1

### PR DESCRIPTION
**Pull request by stefanleoussis**
### What this PR does:
- Implements Socket.io: Users will be able to emit messages (bi-directionally) to other users connected to the same server. 

### Screenshots / Videos:
Screen Shot 2021-06-02 at 8 27 06 PM

### Any information needed to test this feature:
   1. Open a tab in two different Browsers or Two incognito Tabs (a precaution to be sure that cookies are not shared) and head to the localhost that the client is running on (npm start)
  2. Currently the translation feature is not fully implemented so toggle the Original Language toggle to ON
  3. Begin sending messages between the two different clients
  
### Any thoughts or concerns with the current functionality):
  - User id's that are emitted on a message sent will be hardcoded for now (because we are not yet pulling from the database)
- In the Image the input bar is half the width of what it should be, this should not occur when you view the UI (this a separate issue with my system)
- Users are unidentifiable on the UI but this will change later as the database routes will be fully implemented.

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
     "eslint": "^7.26.0",
     "formik": "^2.2.6",
     "react": "^17.0.1",
+    "react-bootstrap": "^1.6.0",
```
Comment (dproc96) [inline client/package.json:21]: Definitely not ideal to import a large library like bootstrap just to handle form logic, you can just use a basic HTML form tag instead.

**Review by dproc96 — COMMENTED** [review]
Screen Shot 2021-06-03 at 8 31 13 AM
I noticed some warnings in the console, can you look into those?

File: client/package.json (hunk 1 above)
Comment (stefanleoussis) [reply client/package.json]: Yea definitely, I was using it for testing but now it does not make sense anymore: Removed

**Comment by stefanleoussis** [comment]
Hi, I resolved most of the warnings except for the protected route. Lei created this Route so I have been working with him to try and solve it. It doesn't seem to be an immediate issue so either he will commit this fix on another branch or we will solve it in the next hour. If you are able to determine what the type is please let us know. Thank you. @dproc96

File: client/src/components/ProtectedRoute.tsx [hunk 2]
```diff
@@ -4,14 +4,16 @@ import CircularProgress from '@material-ui/core/CircularProgress';
 import { useAuth } from '../context/useAuthContext';
 
 interface Props {
+  // eslint-disable-next-line
```
Comment (dproc96) [inline client/src/components/ProtectedRoute.tsx:7]: Lei's PR to fix these looks good so you can revert the commit that suppressed these!

**Review by dproc96 — APPROVED** [review]
This looks good to go, thanks for fixing those warnings

File: client/src/components/ProtectedRoute.tsx (hunk 2 above)
Comment (stefanleoussis) [reply client/src/components/ProtectedRoute.tsx]: Awesome, Commit Reverted 👍

**Pull request merged**